### PR TITLE
fix: add vocalized details on product rating (#13576)

### LINKED ISSUE #13576: Product rating and number of reviews not vocalized
**Describe the bug**

- Product rating is not being narrated properly. Currently not receiving keyboard focus and this is OK, but if the user manages to get there by "Read Mode", is also unable to get there as it is set as "disabled".
- The number of reviews doesn't get narrated as such, but only as a number. Currently not receiving keyboard focus and this is OK, but if the user manages to get there by "Read Mode", it only gets vocalized as "3, text element".

**Tell us the version of Spartacus**
- Library version: 4.0.0

**To Reproduce**
1. Open any Screen Reader (VO, NVDA, JAWS)
2. Go to a Product Details page with reviews (E.g. HDR-CX105E Red)
3. Tab to "Show Reviews" Link
4. Switch to "Read Mode" in Screen Reader (navigate with aux key + arrows instead of Tab key).
5. Listen to SR narration.

**Expected behavior**
- When browsing in "read mode" Screen Reader vocalizes rating as: _"Rated 3.5 out of 5 stars"_
- When browsing in "read mode" Screen Reader vocalizes the number of reviews as: _"19 reviews"_
- When focusing on the "Show reviews" button Screen Reader vocalizes: _"Show reviews, button, 19 reviews, Rated 3.5 out of 5 stars"_
- Use `aria-label`, `aria-describedby`, `cx-visually-hidden`

**Screenshots**
Screen Shot 2021-08-30 at 7 56 58 PM

**Potential Solution**
(ID names are only examples, please follow standard naming convention for IDs.)

Screen Shot 2021-08-31 at 8 02 19 AM

### PR DESCRIPTION
**Pull request by BenjaminEmiliani**
Closes #13576

### CODE REVIEW

**Review by developpeurweb — DISMISSED** [review]
LGTM! Works on VO, JAWS and NVDA

File: projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts [hunk 1]
```diff
@@ -84,4 +84,11 @@ export class ProductIntroComponent implements AfterContentChecked {
       }
     }
   }
+
+  /**
+   * Round average rating to the nearest tenth
+   */
+  roundAverageRating(rating: number): number {
+    return Math.round(rating * 10) / 10;
```
Comment (saad-mo) [inline projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts]: maybe we can use built-in angular [DecimalPipe](https://angular.io/api/common/DecimalPipe#parameters)

File: projects/storefrontlib/cms-components/product/product-intro/product-intro.component.html [hunk 2]
```diff
@@ -8,6 +8,14 @@
       *ngIf="reviewsTabAvailable | async"
       class="btn btn-link cx-action-link"
       (click)="showReviews()"
+      [attr.aria-label]="
+        'productSummary.showReviewsDetailed'
+          | cxTranslate
+            : {
+                rating: roundAverageRating(product.averageRating),
```
Comment (saad-mo) [inline projects/storefrontlib/cms-components/product/product-intro/product-intro.component.html]: we can use the built in angular DecimalPipe and not have to create a new function

File: projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts (hunk 1 above)
Comment (BenjaminEmiliani) [reply projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts]: Removed func, used DecimalPipe instead

File: projects/storefrontlib/cms-components/product/product-intro/product-intro.component.html (hunk 2 above)
Comment (BenjaminEmiliani) [reply projects/storefrontlib/cms-components/product/product-intro/product-intro.component.html]: Removed func, used DecimalPipe instead

**Comment by saad-mo** [comment]
The code changes to improve the narration of the `show reviews` button looks good. however, when you press the button (enter or space) the review tab gets activated and the page scrolls down to the reviews section. in the SR world, there is no indication that this is happening. 

@developpeurweb any thoughts on this? I don't know if there is a separate issue for this already.

**Comment by BenjaminEmiliani** [comment]
> The code changes to improve the narration of the `show reviews` button looks good. however, when you press the button (enter or space) the review tab gets activated and the page scrolls down to the reviews section. in the SR world, there is no indication that this is happening.
> 
> @developpeurweb any thoughts on this? I don't know if there is a separate issue for this already.

Spoke with @developpeurweb, need to create separate ticket for this and to be done after Caine is done refactoring #11637

**Review by saad-mo — APPROVED** [review]

**Pull request merged**